### PR TITLE
Review

### PR DESCRIPTION
**Pull request by ian-antking**

### CODE REVIEW

File: __tests__/pet.test.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
+const Pet = require('../src/pet');
+
+describe('constructor', () =>{
+    let pet
+    beforeEach(() => {
+        pet = new Pet('Daphne');
```
Comment (ian-antking) [inline __tests__/pet.test.js]: Good use of the `beforeEach()`

File: __tests__/pet.test.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    it ('has an initial hunger of 0', () => {
+        expect(pet.hunger).toEqual(0);
+    });
+
+    it('has an initial fitness of 10', () => {
+        expect(pet.fitness).toEqual(10)
+    });
+});
+
+describe('growUp function', () =>{
```
Comment (ian-antking) [inline __tests__/pet.test.js]: If your put all of your describe blocks in one big `describe`, then you would be able to use the same `beforeEach` for all of your tests.

File: __tests__/pet.test.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    it('walk increases fitness by 4', () => { 
+        pet.fitness = 4
+        pet.walk()
+        expect(pet.fitness).toEqual(8);
+    });
+
+    it('walk cannot increase fitness higher than 10', () => {
+        pet.fitness = 10
+        pet.walk()
+        expect(pet.fitness).toEqual(10);
```
Comment (ian-antking) [inline __tests__/pet.test.js]: It might be worth adding a few tests for cases where fitness is only one less than maximum before it's called.

File: __tests__/pet.test.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+describe('adopt a child method', () => { 
+    let parent
+    let child
+    beforeEach(() => {
+        parent = new Pet('Daphne');
+        child = new Pet('Denzel');
+    });
+
+    it('returns 2 instances', () =>{
```
Comment (ian-antking) [inline __tests__/pet.test.js]: This isn't necessary, we already know that the `Pet` constructor returns an `Object`.

File: src/pet.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    this.name = name;
+    this.age = startingAge;
+    this.hunger = startingHunger;
+    this.fitness = maximumFitness
+    this.children = []
+}
+
+Pet.prototype = {
+    get isAlive () {
+            const alive =  this.age < 30 && this.hunger < 10 && this.fitness > 0;
```
Comment (ian-antking) [inline src/pet.js]: You could just return this line rather than assign it to a variable.

File: src/pet.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        this.hunger += 5
+        this.fitness -= 3
+    },
+
+    walk: function() {
+        if (!this.isAlive) {
+            throw new Error ('Your pet is no longer alive');
+        };
+
+       (this.fitness < 7 ? this.fitness += 4 : this.fitness = maximumFitness)
```
Comment (ian-antking) [inline src/pet.js]: Excellent use of a ternary!

File: src/pet.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    },
+
+    feed: function() {
+        if (!this.isAlive) {
+            throw new Error ('Your pet is no longer alive');
+        };
+
+        (this.hunger > 2 ? this.hunger -= 3 : this.hunger = minimumHunger)
```
Comment (ian-antking) [inline src/pet.js]: Your logic is really good here, but there are a few magic numbers.

Comment (ian-antking) [reply src/pet.js]: Instead, you could try something like `this.hunger - 3 < minimumHunger ? foo : bar`
